api/envoy/extensions/network/dns_resolver/cares/v3/cares_dns_resolver.proto

```diff
@@ -39,6 +40,9 @@ message CaresDnsResolverConfig {
   // provide IPv4 addresses.
   bool filter_unroutable_families = 4;
 
+  // Treat ARES_ENODATA and ARES_ENOTFOUND as valid responses and propagate the "no records" result back to the caller.
+  bool accept_nodata = 5;
```

Comment:
IMO this is the way it should work by default and this is a bug, however, this is a high risk change, and I think we should runtime guard it (feature flag). WDYT?

Reply:
+1, this is a high risk change.

I wanted to add a runtime guard but then decided to make it a configurable thing by changing the proto and make it default off because:

1. If this breaks someone then it probably implies their resolver/recursive might not behave correctly and it is probably a long term thing to fix the dns infra, meanwhile they will need to have a way to turn it off.
2. IMO runtime guard is a temp thing and we'll flip it on and remove the guard at some time.

That said, I'm totally fine to add a guard and **make it effective only if both the guard and the config are turned on**, do we want that or just a runtime guard w/o the config option (i.e. change the default behavior but guard it with a flag)?

Reply:
I think I would change the default behavior, clearly document it, and add the runtime flag for this behavior change. We can add a config option layer if someone complains about it because they have to flip the flag. APIs are forever and I would rather not add a new API unless we really need to. Thank you!

/wait

Reply:
SGTM, thanks for the clarification! Can you tell me a little bit more about the "config option layer" you mentioned above? I'm afraid this is something I'm not familiar with (unless it means the proto change we've already been doing here :grinning: )

Reply:
Sorry s/layer/later :)